### PR TITLE
Couple of fixes

### LINKED ISSUE #116: Children and watch window ordering
image image

These should be sorted as:
* no leading underscore (`public`)
* single leading underscore (`_private`)
* double leading underscore (`__uninheritable`)
* dunder names (`__file__`)

### LINKED ISSUE #165: Copy details on exception is missing source and stack trace
Visual Studio

```
raise NameError('name')
```

Hit Copy Details in the popup, it will give you this:
```

  Message=NameError('name',)
  Source=
  StackTrace:

```

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #165
Fixes #116

### CODE REVIEW

**Comment by karthiknadig** [comment]
@int19h I used `sort` instead of `sorted` since I am using a list and sort works in-place. Will that work?

File: ptvsd/wrapper.py [hunk 1]
```diff
@@ -737,19 +739,43 @@ def on_variables(self, request, args):
...
+
+        def get_sort_key(o):
+            return o['name']
+
+        variables.sort(key=get_sort_key)
+        single_underscore.sort(key=get_sort_key)
+        double_underscore.sort(key=get_sort_key)
+        dunder.sort(key=get_sort_key)
+
+        variables = variables + single_underscore + double_underscore + dunder
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: I'd move all of this into a separate class to track and sort the variables.
If tomorrow we need to sort private variables or member variables, we need to modify this method again, when logic of sorting the list should be outside.

Comment (DonJayamanne) [reply ptvsd/wrapper.py]: This stuff or sorting and tracking variables should be in its own class, something that can be easily unit tested in isolation.

Comment (DonJayamanne) [reply ptvsd/wrapper.py]: I think we'll need unit tests for this sorting rule.

Comment (karthiknadig) [reply ptvsd/wrapper.py]: I will move this to a class before merging.

Comment (DonJayamanne) [reply ptvsd/wrapper.py]: :+1:

File: ptvsd/wrapper.py [hunk 2]
```diff
@@ -739,10 +773,7 @@ def on_variables(self, request, args):
         except AttributeError:
             xvars = []
 
-        variables = [] # variables that do not begin with underscores
-        single_underscore = [] # variables that begin with underscores
-        double_underscore = [] # variables that begin with two underscores
-        dunder = [] # variables that begin and end with double underscores
+        variables = VariablesSorter()
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: 👍

**Pull request merged**